### PR TITLE
Add SR-IOV support to nmcli module

### LINKED ISSUE #9122: Add SR-IOV support to nmcli module
### Summary

Allow the SR-IOV settings documented here to be configured via Ansible nmcli module: https://networkmanager.pages.freedesktop.org/NetworkManager/NetworkManager/settings-sriov.html

### Issue Type

Feature Idea

### Component Name

nmcli

### Additional Information

_No response_

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by ianb-mp**
##### SUMMARY

Allow the SR-IOV settings documented here to be configured via Ansible nmcli module: https://networkmanager.pages.freedesktop.org/NetworkManager/NetworkManager/settings-sriov.html

Closes #9122 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

nmcli

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by ianb-mp** [comment]
All lint checks are passing, but I was unable to make the unit test work. I've run out of time unfortunately.

File: plugins/modules/nmcli.py [hunk 1]
```diff
@@ -1058,6 +1058,38 @@
...
+                description:
+                    - Select the eswitch mode of the device.
+                type: int
+            total-vfs:
+                description: Number of virtual functions to create. Consult your NIC documentation for the maximum number of VFs supported.
+                type: int
+            vfs:
+                description:
+                    - 'Virtual function descriptors in the form: "INDEX [ATTR=VALUE[ ATTR=VALUE]...]".'
+                    - Multiple VFs can be specified using a comma as separator e.g. "2 mac=00:11:22:33:44:55 spoof-check=true,3 vlans=100".
```
Comment (felixfontein) [inline plugins/modules/nmcli.py:1091]:
```suggestion
                    - 'Virtual function descriptors in the form: V(INDEX [ATTR=VALUE[ ATTR=VALUE]...]).'
                    - Multiple VFs can be specified using a comma as separator, for example V(2 mac=00:11:22:33:44:55 spoof-check=true,3 vlans=100).
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments below.

**Comment by ianb-mp** [comment]
hello @felixfontein , fyi I add the changes you suggested and also fixed the failing CI.

**Review by russoz — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@ianb-mp thanks for your contribution!
@russoz thanks for reviewing!

**Pull request merged**